vm/src/org/mozartoz/truffle/nodes/call/SelfTailCallThrowerNode.java

```diff
@@ -0,0 +1,30 @@
+package org.mozartoz.truffle.nodes.call;
+
+import org.mozartoz.truffle.nodes.OzNode;
+import org.mozartoz.truffle.runtime.OzArguments;
+import org.mozartoz.truffle.runtime.SelfTailCallException;
+
+import com.oracle.truffle.api.frame.VirtualFrame;
+import com.oracle.truffle.api.nodes.ExplodeLoop;
+
+public class SelfTailCallThrowerNode extends OzNode {
+
+	@Children final OzNode[] arguments;
+
+	public SelfTailCallThrowerNode(OzNode[] arguments) {
+		this.arguments = arguments;
+	}
+
+	public Object execute(VirtualFrame frame) {
+		replaceArguments(frame);
+		throw new SelfTailCallException();
```

Comment:
(the PEA should take care of it but it will help in interpreter and simplifies the graph earlier).